homeassistant/components/snmp/sensor.py

```diff
@@ -79,6 +81,7 @@
         vol.Optional(CONF_PRIV_PROTOCOL, default=DEFAULT_PRIV_PROTOCOL): vol.In(
             MAP_PRIV_PROTOCOLS
         ),
+        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.integer,
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
We generally also don't allow timeout options but want integrations to use a sane default.

Reply:
I was just trying to add the timeout parameter and wasn't thinking of converting the whole integration.  :)  That'll be a much bigger change and not sure I have time to tackle that right now.  So is this completely blocked now unless I can convert the integration away from the yaml?  I don't really own this integration as I was just trying to add this option in there since it failed for me completely without it.

For the timeout itself, there is no default specified in this integration at all.  It happens to be using the python snmp library that has a default on the UdpTransportTarget class of 1 second.  1 second does not work for a lot of slower devices (e.g. with the default, my QNAP NAS never completes the SNMP get within 1 second).  Since there is nothing in the integration to specify a timeout (default or not), even if we have to add a "sane default" we will still have to add the code to specify the value.  And for something like this SNMP integration, I'm not sure there is a "sane default" that will fit people's needs (unless we make the timeout bigger than most folks need), so it seemed to me a better option to allow users to override it (even if there is a default).  My PR kept the default as 1 second so it doesn't change the behaviour on folks that may have been depending on the previous code implicitly doing 1 second timeout.  But open to suggestion on how we can fix this a different way if not providing a timeout override option.

Reply:
What would you set the timeout to?

Reply:
I have one that times out at 1s but is ok with 2s but its not critical for me so I never really cared.

For integrations with rapid scan intervals < 1m, I usually suggest 80% of the scan interval / number of questions made in the scan interval.  So 8s timeout if it does one query or 4s if it does two queries.

Reply:
If we think 8s is the right default I can update the PR with that as the default, but it is probably still good to provide the override option, or are we saying we don't want a user overridable option and the integration just use the 8s?

Reply:
You would need to do the config flow conversion to unblock adding an option. If you don't want to do that, setting the timeout to 8s seems fine to me.

Reply:
Ok, I have taken out adding the option to override and just setting the default timeout to 8s for now.  I can take a look at the conversion of the config flow when I get more time but will open a separate PR for it to keep it clean.

Reply:
We would still be hesitant to add this option in an options flow. We would appreciate if you want to convert the integration to use a config flow but we can't guarantee now that a timeout option will be accepted.